src/util/nextThought.js

```diff
@@ -116,19 +118,20 @@ const nextInContextView = (value, rank, path, rankedContext, contextChain, ignor
   if (rankedContext.length === 0 || path.length === 0) return null
 
   const context = pathToContext(rankedContext)
+  const firstChild = perma(() => firstChildOfContextView(path, thoughtIndex))
 
   // if the focus is on a thought with context view open, move it into context view - jump in
-  if (isValidContextView(pathToContext(path))) {
+  if (!ignoreChildren && isValidContextView(path) && firstChild()) {
     const currentThought = head(path)
-    const firstChild = firstChildOfContext(path, thoughtIndex)
     // jump out if there are no context children
-    return !ignoreChildren && firstChild
-      ? { nextThoughts: [...firstChild, currentThought], contextChain }
-      : nextInContextView(value, rank, path, contextOf(rankedContext), contextOf(contextChain))
+    return {
+      nextThoughts: [...firstChild(), currentThought],
+      contextChain
+    }
```

Comment:
When there are no context children, we can allow it to move to the default case. This simplifies this case.